provider/pkg/provider/provider.go

```diff
@@ -805,54 +815,47 @@ func (p *cfnProvider) Create(ctx context.Context, req *pulumirpc.CreateRequest)
 	}
 
 	resourceToken := string(urn.Type())
-	var spec schema.CloudAPIResource
-	var hasSpec bool
-	var cfType string
+	var id *string
+	var outputsMap map[string]interface{}
+	var createErr error
 	var payload map[string]interface{}
-	switch resourceToken {
-	case schema.ExtensionResourceToken:
-		// For a custom resource, both CF type and inputs shape are defined explicitly in the SDK.
-		inputsMap := resourcex.Decode(inputs)
-		if v, has := inputsMap["type"].(string); has {
-			cfType = v
-		} else {
-			return nil, errors.New("no 'type' property in extension resource inputs")
-		}
-		if v, has := inputsMap["properties"].(map[string]interface{}); has {
-			payload = v
-		} else {
-			return nil, errors.New("no 'properties' property in extension resource inputs")
-		}
-	default:
-		spec, hasSpec = p.resourceMap.Resources[resourceToken]
+	if customResource, ok := p.customResources[resourceToken]; ok {
+		var outputs resource.PropertyMap
+		id, outputs, createErr = customResource.Create(ctx, urn, inputs)
+		outputsMap = resourcex.Decode(outputs)
```

Comment:
It looks like we converted a map to PropertyMap inside the resource then back to map here.

Reply:
Yeah, still a little confused on where we should be using more of `PropertyMap`'s features in the main provider, but this is a pretty minor overhead to ensure the interface provides the richer type system.